### PR TITLE
Dpl680 ruby 3 upgrade tests passing

### PR DESCRIPTION
**Pull request by emrojo**
Closes DPL680

#### Changes proposed in this pull request

- Ruby 3 support

#### Instructions for Reviewers

_[All PRs] - Confirm PR template filled_  
_[Feature Branches] - Review code_  
_[Production Merges to `main`]_  
 &nbsp; &nbsp; \- _Check story numbers included_  
 &nbsp; &nbsp; \- _Check for debug code_  
 &nbsp; &nbsp; \- _Check version_

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -39,9 +39,10 @@ Rails/SkipsModelValidations:
     - "features/support/step_definitions/*.rb"
 
 Style/Documentation:
-  Exclude:
-    - "test/**/*"
-    - "spec/**/*"
+  Enabled: false
+  #Exclude:
```
Comment (stevieing) [inline .rubocop.yml:43]: can you remove the comments?

File: Gemfile [hunk 2]
```diff
@@ -28,7 +28,7 @@ group :default do
 
   # Provides bulk insert capabilities
   gem 'activerecord-import'
-  gem 'record_loader'
+  gem 'record_loader', git: 'https://github.com/sanger/record_loader', branch: 'dpl680_upgrade_ruby3'
```
Comment (stevieing) [inline Gemfile:31]: What are you going to do with this. Does it need sorting once this is moved to production?

File: app/helpers/application_helper.rb [hunk 3]
```diff
@@ -310,3 +310,27 @@ def legacy_javascript_tag
   end
 end
 # rubocop:enable Metrics/ModuleLength
+
+# error_messages_for method was deprecated, however lots of the tests depend on the message format it
```
Comment (stevieing) [inline app/helpers/application_helper.rb:314]: Does this exist only for tests?

**Review by stevieing — APPROVED** [review]
Nice piece of work. Glad you managed to fix the cucumbner issues.
Just a quick browse with a couple of questions.
Didn't even bother looking at Cucumber changes.
Noticed there are a few tests that are skipped. Maybe we should remove them?

File: .rubocop.yml (hunk 1 above)
Comment (emrojo) [reply .rubocop.yml:43]: Ok

File: Gemfile (hunk 2 above)
Comment (emrojo) [reply Gemfile:31]: No, I'll change it

File: app/helpers/application_helper.rb (hunk 3 above)
Comment (emrojo) [reply app/helpers/application_helper.rb:314]: No, it is replacing a method that has been deprecated and no longer worked. It is in use in the app now to display errors.

**Pull request merged**